Assignment2/carCleaningService.rb

```diff
@@ -0,0 +1,144 @@
+require 'time'
+
+# Describes the schedule by which the service operates
+module Schedule
+  START_WEEK_DAY = 1 # Monday
+  END_WEEK_DAY = 5 # Friday
+  START_TIME_HOUR = 8
+  END_TIME_HOUR = 17
+
+  def self.open?(time)
+    open_time?(time) && open_day?(time)
+  end
+
+  def self.open_time?(current_time)
+    current_time.hour > START_TIME_HOUR && current_time.hour < END_TIME_HOUR
+  end
+
+  def self.open_day?(current_day_of_week)
+    current_day_of_week.wday > START_WEEK_DAY && current_day_of_week.wday < END_WEEK_DAY
+  end
+end
+
+# Clients of the car wash service. Will be updated once their car is ready for pick up and then pick a time to come for it
+class User
+  attr_reader :name, :car, :time_of_arrival, :time_of_pickup
+
+  include Schedule
+
+  def initialize(name, car, time_of_arrival)
+    @name = name
+    @car = car
+    @time_of_arrival = time_of_arrival
+  end
+
+  def update
+    puts "#{@name} was notified"
+    set_pick_up_time
+    puts "#{@name} will be there to pick car up #{@time_of_pickup}"
+  end
+
+  def set_pick_up_time
+    # Sets a random time after the current one (in HOURS)
+    @time_of_pickup = Time.new + rand(1..100) * 10 * 60 * 60
+    # Check if the set time is in conformity with the schedule of the car wash
+    if Schedule.open?(@time_of_pickup)
+      @time_of_pickup
+    else
+      set_pick_up_time
+    end
```

Comment:
another way of writing this with one liner:

`Schedule.open?(@time_of_pickup) ? time_of_pickup : set_pick_up_time`